packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -46,6 +46,9 @@ const dynamicImportPrefixRE = /import\s*\(/
 const optimizedDepChunkRE = /\/chunk-[A-Z\d]{8}\.js/
 const optimizedDepDynamicRE = /-[A-Z\d]{8}\.js/
 
+const dynamicImportTreeshakenRE =
+  /(\b(const|let|var)\s+(\{[^}.]+\})\s*=\s*await\s+import\([^)]+\))|(\(\s*await\s+import\([^)]+\)\s*\)(\??\.[^;[\s]+)+)|\bimport\([^)]+\)(\.then\([^{]*\{([^}]+)\})/g
```

Comment:
The below case cannot be supported.

```ts
import('./deps.js')
  .then({ a } => {})
```

You need to add `\s*` before `.then`

IMO, using regexp to handle this might break many edge-case

Reply:
Yeah, regexp can hardly cover all the edge-cases, but it seems there's no better way to support dynamic import treeshaking within current logic.

IMO, we could support the common usages of dynamic import for now, and then patch the related issues in the future